diff --git a/src/include/nodes/memnodes.h b/src/include/nodes/memnodes.h
--- a/src/include/nodes/memnodes.h
+++ b/src/include/nodes/memnodes.h
@@ -130,6 +130,7 @@ typedef struct MemoryContextData
 	MemoryContext nextchild;	/* next child of same parent */
 	const char *name;			/* context name */
 	const char *ident;			/* context ID if any */
+	const void *libpg_query_plpgsql_catalog;
 	MemoryContextCallback *reset_cbs;	/* list of reset/delete callbacks */
 } MemoryContextData;
 
diff --git a/src/backend/utils/mmgr/mcxt.c b/src/backend/utils/mmgr/mcxt.c
--- a/src/backend/utils/mmgr/mcxt.c
+++ b/src/backend/utils/mmgr/mcxt.c
@@ -1088,6 +1088,7 @@ MemoryContextCreate(MemoryContext node,
 	node->prevchild = NULL;
 	node->name = name;
 	node->ident = NULL;
+	node->libpg_query_plpgsql_catalog = NULL;
 	node->reset_cbs = NULL;
 
 	/* OK to link node into context tree */
